app/src/ErrorHandler/ViewRenderer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ErrorHandler;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Spiral\Boot\DirectoriesInterface;
+use Spiral\Http\ErrorHandler\RendererInterface;
+use Spiral\Http\Header\AcceptHeader;
+use Spiral\Views\ViewsInterface;
+
+class ViewRenderer implements RendererInterface
+{
+    private const GENERAL_VIEW = 'exception/error.dark.php';
+    private const VIEW = 'exception/%s.dark.php';
+
+    public function __construct(
+        private readonly ViewsInterface $views,
+        private DirectoriesInterface $dirs,
+        private readonly ResponseFactoryInterface $responseFactory
+    ) {
+    }
+
+    public function renderException(Request $request, int $code, string $message): ResponseInterface
+    {
+        $acceptItems = AcceptHeader::fromString($request->getHeaderLine('Accept'))->getAll();
+        if ($acceptItems && $acceptItems[0]->getValue() === 'application/json') {
+            return $this->renderJson($code, $message);
+        }
+
+        return $this->renderView($code, $message);
+    }
+
+    private function renderJson(int $code, string $message): ResponseInterface
+    {
+        $response = $this->responseFactory->createResponse($code);
+
+        $response = $response->withHeader('Content-Type', 'application/json; charset=UTF-8');
+        $response->getBody()->write(\json_encode(['status' => $code, 'error' => $message]));
+
+        return $response;
+    }
+
+    private function renderView(int $code, string $message): ResponseInterface
+    {
+        $response = $this->responseFactory->createResponse($code);
+        $content = "Error code: {$code}";
+
+        if ($template = $this->findTemplate($code)) {
```

Comment:
I prefer to use 

```php
try {
    $this->views->get($template)->render();
} catch (...)
```

It will compatible with different template engines

Reply:
Yes, agree with you. Done